### PR TITLE
Added validation when database is loaded within the SQLiteKeyInfoManager.

### PR DESCRIPTION
**Pull request by MattDavis00**
This validation checks that:
- The database exists.
- The correct tables exist.
- The `schema_version` is correct.
- All mapping records have the correct `key_info_version`.

There is a future opportunity here to create more unit tests covering:
- Loading an empty database.
- Loading a database with either the `key_mapping` or `kim_metadata` table missing.
- Loading a database with an incorrect `schema_version`.
- Loading a database where some mappings have incorrect/incompatible `key_info_version` values.

Signed-off-by: Matt Davis <matt.davis@arm.com>

### CODE REVIEW

File: src/key_info_managers/sqlite_manager/mod.rs [hunk 1]
```diff
@@ -70,28 +65,171 @@ impl SQLiteKeyInfoManager {
...
+                        .into_string()
+                        .unwrap_or_else(|_| "DB_FILE_PATH_UNKNOWN".to_string()),
+                );
+                error!("{}", error_message);
+                return Err(Error::new(ErrorKind::Other, error_message).into());
+            }
+            // The correct number of tables are present, no-op
+            2 => {}
+            // Shouldn't be possible to reach this. If it does, throw an error
+            _ => {
```
Comment (ionut-arm) [inline src/key_info_managers/sqlite_manager/mod.rs:141]: could you skip `1` above and just let it fall under this? Seems like the code is identical.

File: src/key_info_managers/sqlite_manager/mod.rs [hunk 2]
```diff
@@ -70,28 +65,171 @@ impl SQLiteKeyInfoManager {
...
+            SELECT
+                *
+            FROM
+                key_mapping
+        ",
+        )?;
+        let mut rows = key_info_version_stmt.query(params![])?;
+        while let Some(row) = rows.next()? {
+            let version_number: u8 = row.get("key_info_version")?;
+            if version_number != CURRENT_KEY_INFO_VERSION {
```
Comment (ionut-arm) [inline src/key_info_managers/sqlite_manager/mod.rs:202]: Is it not possible (and I guess this applies to the previous query as well) to just expand the query to ask for entries where the version number has this property (i.e. `!= CURRENT_KEY_INFO_VERSION`)? Would've thought SQL allows us to do that, and then you can just check the row count.

**Review by ionut-arm — COMMENTED** [review]
Thanks for the patch! Don't worry about the build for now 😬

File: src/key_info_managers/sqlite_manager/mod.rs (hunk 2 above)
Comment (MattDavis00) [reply src/key_info_managers/sqlite_manager/mod.rs:202]: Yes 🤦🏼‍♂️ think I was just on autopilot copying from one of the previous SQL statements.

Comment (MattDavis00) [reply src/key_info_managers/sqlite_manager/mod.rs]: Having looked at it a second time. I've added the `WHERE key_info_version != CURRENT_KEY_INFO_VERSION` clause to the SQL statement. I've however chosen not to use the length, this is as the value returned from the query only implements iterator. I would have to call `.count()` which under the hood calls `.next()` for every item.

Opted for this instead:
```
let rows = SQL_QUERY();
// If a mapping exists with the wrong key_info_version, throw an error.
if let Some(row) = rows.next()? {
    .....throw error
}
```

**Review by ionut-arm — APPROVED** [review]
👍🏻 Thanks!

**Pull request merged**